packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -214,12 +215,205 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 	}
 
 	public async getSnapshot(snapshotFetchOptions?: ISnapshotFetchOptions): Promise<ISnapshot> {
-		assert(this.hostPolicy.supportGetSnapshotApi !== true, "api should not be called yet");
-		// This is just temporary as this api is not yet enabled in service policies.
-		return this.fetchSnapshot(
-			this.hostPolicy.snapshotOptions,
-			snapshotFetchOptions?.scenarioName,
+		// If no groupId is specified, then fetch the snapshot with "default" groupId.
+		const groupIds = snapshotFetchOptions?.groupIds ?? [];
+		if (groupIds.length === 0) {
+			groupIds.push(defaultGroupIdForSnapshot);
+		}
+		const { snapshot } = await this.startSnapshotFetch({ ...snapshotFetchOptions, groupIds });
+		return snapshot;
+	}
+
+	private async startSnapshotFetch(
+		snapshotFetchOptions: ISnapshotFetchOptions,
+	): Promise<{ snapshot: ISnapshot; id: string | undefined }> {
+		const hostSnapshotOptions = this.hostPolicy.snapshotOptions;
+		const groupIds = snapshotFetchOptions.groupIds ?? [];
```

Comment:
I think it's second place where we do ?? [].
I'd suggest either most of the layers accept optional array up to the last layer, or vice versa - all layers (except the most top) to make it required.